Neo4j.Driver/Neo4j.Driver.IntegrationTests/Examples.cs

```diff
@@ -18,338 +18,571 @@
+            public ReadWriteTransactionExample(ITestOutputHelper output, StandAloneIntegrationTestFixture fixture)
+                : base(output, fixture)
+            {
+            }
+
+            [RequireServerFact]
+            public void TestReadWriteTransactionExample()
             {
-                var ex = Record.Exception(() =>
+                // When & Then
+                AddPerson("Alice").Should().BeGreaterOrEqualTo(0L);
+            }
+
+            // tag::read-write-transaction[]
+            public long AddPerson(string name)
+            {
+                using (var session = Driver.Session())
                 {
-                    //tag::handle-cypher-error[]
-                    try
-                    {
-                        session.Run("This will cause a syntax error").Consume();
-                    }
-                    catch (ClientException)
-                    {
-                        throw new InvalidOperationException("Something really bad has happened!");
-                    }
-                    //end::handle-cypher-error[]
-                });
-                ex.Should().BeOfType<InvalidOperationException>();
+                    session.WriteTransaction(tx => CreatePersonNode(tx, name));
+                    return session.ReadTransaction(tx => MatchPersonNode(tx, name));
+                }
+            }
+
+            private static void CreatePersonNode(ITransaction tx, string name)
+            {
+                tx.Run("CREATE (a:Person {name: $name})", new {name});
+            }
+
+            private static long MatchPersonNode(ITransaction tx, string name)
+            {
+                var result = tx.Run("MATCH (a:Person {name: $name}) RETURN id(a)", new {name});
+                return result.Single()[0].As<long>();
             }
+            // end::read-write-transaction[]
         }
 
-        [RequireServerFact]
-        public void TransactionCommit()
+        public class ResultConsumeExample : BaseExample
         {
-            using (var session = Driver.Session())
+            public ResultConsumeExample(ITestOutputHelper output, StandAloneIntegrationTestFixture fixture)
+                : base(output, fixture)
+            {
+            }
+
+            // tag::result-consume[]
+            public List<string> GetPeople()
             {
-                //tag::transaction-commit[]
-                using (var tx = session.BeginTransaction())
+                using (var session = Driver.Session())
                 {
-                    tx.Run("CREATE (:Person {name: {name}})",
-                        new Dictionary<string, object> {{"name", "Guinevere"}});
-                    tx.Success();
+                    return session.ReadTransaction(tx =>
+                    {
+                        var result = tx.Run("MATCH (a:Person) RETURN a.name ORDER BY a.name");
+                        return result.Select(record => record[0].As<string>()).ToList();
```

Comment:
Shouldn't `ToList()` be used in `result-retain` example and `result-consume` be a separate thing that show lazy iteration?